src/pip/_internal/commands/show.py

```diff
@@ -30,17 +30,16 @@ class ShowCommand(Command):
       %prog [options] <package> ..."""
     ignore_require_venv = True
 
-    def __init__(self, *args, **kw):
-        # type: (*Any, **Any) -> None
-        super(ShowCommand, self).__init__(*args, **kw)
-        self.cmd_opts.add_option(
+    def add_options(self, cmd_opts):
+        # type: (OptionGroup) -> None
+        cmd_opts.add_option(
             '-f', '--files',
             dest='files',
             action='store_true',
             default=False,
             help='Show the full list of installed files for each package.')
 
-        self.parser.insert_option_group(0, self.cmd_opts)
+        self.parser.insert_option_group(0, cmd_opts)
```

Comment:
Can we something like this here, and all other places we are using `add_options`

```python
    def add_options(self):
    # type: () -> None
    self.cmd_opts.add_option(
        '-f', '--files',
        dest='files',
        action='store_true',
        default=False,
        help='Show the full list of installed files for each package.')

    self.parser.insert_option_group(0, self.cmd_opts)
```

Calling `self.add_options(self.cmd_opts)` in `base_command`, and in subcommands, reassigning that to `cmd_opts` and then adding it to `self.parser.insert_option_group` seems a bit unnecessary to me, when we already have `self.cmd_opts` in the subclasses, seems a bit circular to me.

Reply:
Indeed! That's a very useful simplification!

We can drop the argument in `add_options`, and just use self.cmd_opts directly instead. It also means we won't need to remove any `cmd_opts = self.cmd_opts` or update `self.cmd_opts.add_options` calls, which would be a smaller change as well. :)

Reply:
Ah! ok. I'll update it. Thanks for the feedback :)